### DISCUSSION TITLE
Reset admin password directly in database

### DISCUSSION THREAD

**Discussion by smutel** [keycloak/keycloak#31079] [Q&A]
Hello,

The idea is to be able to reset the administrator password to a default password using SQL queries.
So I changed the admin password to a default password and I saved the content of column `secret_data` in table `credential`.
I reinstalled keycloak.
I updated the column `secret_data` in table `credential` with the data generated with default password but it does not work ...
I am not able to login and the keycloak theme is reset to default ...

Any other idea ?
Thanks.

**Comment by shawkins** [comment]
Sorry, there is no good story around built-in admin recovery in recent Keycloak releases. Starting with Keycloak 26, we will support a cli command to create a bootstrap admin user - #9829

You can see some discussion about a database level workaround in https://github.com/keycloak/keycloak/issues/19913#issuecomment-1540782738
